docs/05-tutorials/02-real-time-communication.md

```diff
@@ -0,0 +1,399 @@
+
+:::
+
+You can try out the final app here: [https://pixorama.live](https://pixorama.live)
+
+![Serverpod Insights](/img/tutorial/pixorama/pixorama.png)
+
+## What is Pixorama?
+
+Pixorama is a collaborative drawing app where users can place pixels on a grid to create images together. Imagine two instances of the app running simultaneously - draw a pixel on one and watch it instantly appear on the other. This seamless synchronization happens because each time you draw a pixel, a message is sent to the server, which then broadcasts it to all connected clients.
+
+## Understanding real-time communication
+
+In traditional REST APIs, communication with the server involves sending a request and receiving a response. However, real-time communication requires the server to push updates to clients as they happen. This is commonly achieved using web sockets, which maintain an open connection between the server and client, allowing for continuous data exchange. While web sockets can be tricky, requiring data serialization and connection management, Serverpod simplifies this process.
+
+With the release of Serverpod 2.1, a new feature called [streaming methods](../concepts/streams) was introduced. This feature allows us to return a stream from a server method and call it from our app. Serverpod handles the underlying web socket connection for us. Now, let's get started with building Pixorama.
+
+## Setting up the project
+
+We begin by creating a new project with the `serverpod create` command. Since we don't need to store data in a database, we'll use the Mini version of Serverpod. Serverpod Mini is a lightweight version of Serverpod without a database, advanced logging, and other features - perfect for our needs. Create the project with the command:
+
+```bash
+serverpod create pixorama --mini
+```
+
+Now, let's open the project in VS Code and explore the structure. The server code resides in the `pixorama_server` package. We'll start by creating models - classes that we can serialize and pass between the client and server. Our models will be placed in the `lib/src/models` directory.
+
+## Creating models
+
+First, we remove the `example.spy.yaml` model, as we won't need it. We'll create two new models: `ImageData` and `ImageUpdate`. Place them in the `lib/src/models` directory and call them `image_data.spy.yaml` and `image_update.spy.yaml`.
+
+```yaml
+# lib/src/models/image_data.spy.yaml
+
+class: ImageData
+fields:
+  pixels: ByteData
+  width: int
+  height: int
+```
+
+The `ImageData` model represents the entire image that will be sent to the app when it connects to the server. It stores the image's pixels as ByteData, where each byte represents a pixel. Additionally, it includes the image's width and height.
+
+```yaml
+# lib/src/models/image_update.spy.yaml
+
+class: ImageUpdate
+fields:
+  pixelIndex: int
+  colorIndex: int
+```
```

Comment:
Perhaps it could be a nice plug for the VS code extension to add an ::info block somewhere here and mention that the VS code extension enables intellisense in these yaml files.

Reply:
It's part of the install guide, so I don't think it's needed here.